### PR TITLE
Extract (some) yul syntax tests.

### PR DESCRIPTION
**Pull request by chriseth**

### CODE REVIEW

File: test/libyul/yulSyntaxTests/smok.yul [hunk 1]
```diff
@@ -0,0 +1 @@
+{}
```
Comment (ekpyron) [inline test/libyul/yulSyntaxTests/smok.yul]: Weird filename...

File: test/libyul/yulSyntaxTests/funcitons_multiple_args.yul [hunk 2]
```diff
@@ -0,0 +1 @@
+{ function f(a:u256, d:u256) { } function g(a:u256, d:u256) -> x:u256, y:u256 { } }
```
Comment (ekpyron) [inline test/libyul/yulSyntaxTests/funcitons_multiple_args.yul]: Typo in filename.

File: test/libyul/yulSyntaxTests/instructions.yul [hunk 3]
```diff
@@ -0,0 +1,3 @@
+{ pop }
+// ----
+// ParserError 2314: (6-7): Expected '(' but got '}'
```
Comment (ekpyron) [inline test/libyul/yulSyntaxTests/instructions.yul]: That's surprisingly not the same error message as in the old test...

**Review by ekpyron — COMMENTED** [review]
Went through all of them - the one mismatching error message is a bit weird, but probably nothing to worry about. Apart from that only some typos in filenames.

**Comment by ekpyron** [comment]
Also seems to need a rebase for the Z3 tests to pass (hopefully only that).

File: test/libyul/yulSyntaxTests/smok.yul (hunk 1 above)
Comment (cameel) [reply test/libyul/yulSyntaxTests/smok.yul]: Fun fact: 'smok' means 'dragon' in Polish :)

File: test/libyul/yulSyntaxTests/instructions.yul (hunk 3 above)
Comment (chriseth) [reply test/libyul/yulSyntaxTests/instructions.yul]: Changed the dialect - that did the trick.

File: test/libyul/yulSyntaxTests/variable_declaration_complex.yul [hunk 4]
```diff
@@ -0,0 +1,7 @@
+{
+    function add(a:u256, b:u256) -> c:u256 {}
```
Comment (chriseth) [inline test/libyul/yulSyntaxTests/variable_declaration_complex.yul:2]: Changed this one back from `add1` to `add`

File: test/libyul/yulSyntaxTests/function_calls.yul [hunk 5]
```diff
@@ -0,0 +1 @@
+{ function f(a:u256) -> b:u256 {} function g(a:u256, b:u256, c:u256) {} function x() { g(1:u256, 2:u256, f(3:u256)) x() } }
```
Comment (cameel) [inline test/libyul/yulSyntaxTests/function_calls.yul]: I'd split this (and maybe some others) into separate lines. It's getting hard to parse quickly when the lines start getting long.

That is, unless the intention was to make sure that it still works on a single line, without line breaks - but I don't think that's the case here.

**Review by ekpyron — DISMISSED** [review]
Fine to merge if tests pass!

**Review by cameel — DISMISSED** [review]
I also reviewed them all and everything seems fine. All the things pointed out earlier by @ekpyron seem to have been resolved.

**Comment by chriseth** [comment]
Added some newlines.

**Review by ekpyron — APPROVED** [review]
Reapproving after newlines.

File: test/libyul/yulSyntaxTests/smok.yul (hunk 1 above)
Comment (leonardoalt) [reply test/libyul/yulSyntaxTests/smok.yul]: It also shares the language roots (`smjúga/smaug`) with Smaug from lotr

**Review by cameel — APPROVED** [review]
